### PR TITLE
Add new logic for showing notice on top of the page

### PR DESCRIPTION
**Pull request by henrikvolmer**

### CODE REVIEW

File: src/core/hide.js [hunk 1]
```diff
@@ -1,17 +1,56 @@
 import { addEventListenerOnce } from '../utils/event-listener';
 
+const __slideUp = $el => {
+    let timing = '1000ms ease';
```
Comment (rherwig) [inline src/core/hide.js]: Why let instead of const?

File: src/core/hide.js [hunk 2]
```diff
@@ -1,17 +1,56 @@
...
+    
+    $el.style.webkitTransition = 'initial';
+    $el.style.transition = 'initial';
+    let height = $el.offsetHeight + 'px';
```
Comment (rherwig) [inline src/core/hide.js]: Why let instead of const?

Comment (rherwig) [reply src/core/hide.js]: Use template strings instead of concat with +

File: src/core/hide.js [hunk 3]
```diff
@@ -1,17 +1,56 @@
...
+    $el.style.maxHeight = height;
+    $el.style.overflow = 'hidden';
+    
+    $el.style.webkitTransition = 'max-height ' + timing + ', opacity ' + timing + '';
+    $el.style.transition = 'max-height ' + timing + ', opacity ' + timing + '';
```
Comment (rherwig) [inline src/core/hide.js]: Use template strings on L12+13

File: src/core/hide.js [hunk 4]
```diff
@@ -1,17 +1,56 @@
...
-        document.body.removeChild($background);
-        document.body.removeChild($html);
-
-        if (options.reloadAfter) {
-            location.reload();
+        __remove($background, $html, options);
+    });
+    
+    addEventListenerOnce($content, 'transitionend', () => {
+        if(options.showElementOnTop) {
```
Comment (rherwig) [inline src/core/hide.js]: Missing space after keyword

File: src/core/hide.js [hunk 5]
```diff
@@ -1,17 +1,56 @@
...
+            __remove($background, $html, options);
         }
+        
+        $background.classList.add('cookie-manager--hidden')
     });
-    addEventListenerOnce($content, 'transitionend', () => $background.classList.add('cookie-manager--hidden'));
 
+    if(options.showElementOnTop) {
```
Comment (rherwig) [inline src/core/hide.js]: Missing space after keyword

File: src/core/show.js [hunk 6]
```diff
@@ -48,21 +48,39 @@ const _getBody = options => {
     return $html;
 };
 
+const __appendElementToBody = (onTop, $element) => {
+    if(onTop) {
```
Comment (rherwig) [inline src/core/show.js]: Missing space after keyword

File: src/core/show.js [hunk 7]
```diff
@@ -48,21 +48,39 @@ const _getBody = options => {
...
 
-    document.body.appendChild($background);
-    document.body.appendChild($html);
-
-    $background.style.display = 'inherit';
-
+    __appendElementToBody(showTop, $html);
+    __appendElementToBody(showTop, $background);
+    
+    if(!showTop) {
```
Comment (rherwig) [inline src/core/show.js]: Missing space after keyword

File: src/core/show.js [hunk 8]
```diff
@@ -48,21 +48,39 @@ const _getBody = options => {
...
+        $background.style.display = 'inherit';
+    }
+    
     if ($content) {
         addEventListenerOnce($background, 'transitionend', () =>
             $content.classList.add('content--visible')
         );
+        
+        if(showTop) {
```
Comment (rherwig) [inline src/core/show.js]: Missing space after keyword

File: src/core/hide.js [hunk 9]
```diff
@@ -1,17 +1,56 @@
...
+        $el.style.removeProperty('transition');
+        $el.removeEventListener(transitionEvent('end'), endSlideDown);
+    };
+    requestAnimationFrame(() => {
+        $el.style.maxHeight = '0';
+        $el.style.opacity = '0';
+    });
+}
+
+const __remove = ($background, $html, options) => {
```
Comment (rherwig) [inline src/core/hide.js:25]: Why double lodash before function name?

File: src/core/show.js (hunk 6 above)
Comment (rherwig) [reply src/core/show.js]: Why double lodash before function name?

**Review by rherwig — COMMENTED** [review]
Review failed with code style issues

File: src/core/show.js (hunk 6 above)
Comment (henrikvolmer) [reply src/core/show.js]: This is the code style of this module. Double lodash shows that is a private not exported function

**Review by rherwig — APPROVED** [review]

**Pull request merged**